### ISSUE TITLE
Add a directive for treating a command as an alias of another command

### ISSUE THREAD

**Issue by thecaralice** [koalaman/shellcheck#3060]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related (the closest is #1132)

In the example provided below, `info` essentially behaves like `printf` in terms of argument processing, so I would expect to have an option to say "info is basically printf with a different name" to shellcheck

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

name="alice"

info() {
    printf '\x1b[2m'
    # shellcheck disable=SC2059 # intentional
    printf "$@"
    printf '\x1b[m'
}

info "hello $name"
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
```
Line 12:
    info "$@"
           ^-- SC2059 (info): Don't use variables in the printf format string. Use printf '..%s..' "$foo".
```
